### PR TITLE
Features - fetching data from Rest API  ,Profile page

### PR DESCRIPTION
**Pull request by mehul-jain**

### CODE REVIEW

File: app/src/main/java/com/example/admin/showdata/DownloadData.java [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Created by admin on 15-02-2018.
+ */
+
+public class DownloadData {
+    HttpURLConnection connection;
+    List<ContentModel> data_list;
```
Comment (dej0e) [inline app/src/main/java/com/example/admin/showdata/DownloadData.java]: Naming convention again. Follow it.

File: app/src/main/java/com/example/admin/showdata/RecyclerViewAdapter.java [hunk 2]
```diff
@@ -40,18 +41,30 @@ public void onBindViewHolder(View_Holder holder, int position) {
 
     @Override
     public int getItemCount() {
-        return content_list.size();
+            return content_list.size();
     }
 
     @Override
```
Comment (dej0e) [inline app/src/main/java/com/example/admin/showdata/RecyclerViewAdapter.java]: Do not override methods which you aren't going to use.

File: app/src/main/java/com/example/admin/showdata/RecyclerViewAdapter.java [hunk 3]
```diff
@@ -40,18 +41,30 @@ public void onBindViewHolder(View_Holder holder, int position) {
...
     }
 
-    public void insert(int position,Content_model content)
+    // insert a new data to list
+    public void insert(int position,ContentModel content)
     {
         content_list.add(position,content);
         notifyItemInserted(position);
     }
+    public class ViewHolder extends RecyclerView.ViewHolder {
```
Comment (dej0e) [inline app/src/main/java/com/example/admin/showdata/RecyclerViewAdapter.java]: Why public?

File: app/src/main/java/com/example/admin/showdata/DownloadData.java [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+
+
+    public List<ContentModel> getdatafromurl(String passed_url){
```
Comment (VeneetReddyK) [inline app/src/main/java/com/example/admin/showdata/DownloadData.java]: Naming convention violation. Never use snake case in Java.

File: app/src/main/java/com/example/admin/showdata/DownloadData.java [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+                    ContentModel content_model = new ContentModel();
+                    // getting  individual object
+                    JSONObject jsonObject = jsonArray.getJSONObject(i);
+                    content_model.setTitle(jsonObject.getString("title"));
+                    content_model.setDescription(jsonObject.getString("description"));
+                    content_model.setImageId(R.drawable.movie_night);
+                    data_list.add(content_model);
+                }
+
+            } catch (Throwable t) {
```
Comment (VeneetReddyK) [inline app/src/main/java/com/example/admin/showdata/DownloadData.java]: Don't catch generic `Exception`s/`Throwable`s. Keep them as specific as possible. Reasons mentioned [here](https://source.android.com/setup/code-style#dont-catch-generic-exception).

File: app/src/main/java/com/example/admin/showdata/DownloadData.java [hunk 6]
```diff
@@ -0,0 +1,72 @@
...
+                Log.d("Result String", "Could not parse malformed JSON: \"" + resultStr + "\"");
+            }
+        }catch( Exception e) {
+            e.printStackTrace();
```
Comment (VeneetReddyK) [inline app/src/main/java/com/example/admin/showdata/DownloadData.java]: Generic exception again. Also use the `Log` class to print to Logcat. It make the Logcat more searchable with TAGs, color highlighting, etc. In this case to print the stack-trace you should be using:

`Log.e(TAG, "Exception while fetching data: ", e);` Note `e`'s override accepts an exception too.

File: app/src/main/java/com/example/admin/showdata/DownloadData.java [hunk 7]
```diff
@@ -0,0 +1,72 @@
...
+
+    public List<ContentModel> getdatafromurl(String passed_url){
+        StringBuilder result = new StringBuilder();
+        try {
+            // getting url
+            URL url = new URL(passed_url);
+            // setting connection
+            connection = (HttpURLConnection) url.openConnection();
+            // setting stream
+            InputStream in = new BufferedInputStream(connection.getInputStream());
```
Comment (VeneetReddyK) [inline app/src/main/java/com/example/admin/showdata/DownloadData.java]: `InputStream` and `BufferedReader` are not closed after its use. This causes a leak of resources used by them. Use [try-with-resources](https://docs.oracle.com/javase/tutorial/essential/exceptions/tryResourceClose.html) for both. They close automatically when your done with them.

**Pull request merged**